src/julia-syntax.scm

```diff
@@ -2242,7 +2242,7 @@
       ,.(map (lambda (v r) `(= ,v (call (top length) ,r))) lengths rv)
       (scope-block
        (block
-        (= ,result (call (core Array) ,atype ,@lengths))
+        (= ,result (call (curly Array ,atype) ,@lengths))
```

Comment:
can also add `,(length lengths)` to the `(curly`, if I'm not mistaken, to splice in the number of dimensions also